rust/chains/tw_polkadot/src/scale.rs

```diff
@@ -0,0 +1,442 @@
+            },
+            _ => {
+                let bytes_needed = 8 - self.0.leading_zeros() / 8;
+                out.reserve(bytes_needed as usize);
+                out.push(0b11 + ((bytes_needed - 4) << 2) as u8);
+                let mut x = self.0;
+                for _ in 0..bytes_needed {
+                    out.push(x as u8);
+                    x >>= 8;
+                }
+            },
+        }
+    }
+}
+
+// TODO: implement U128 support (we don't have it yet in tw_number)
+
+impl ToScale for Compact<U256> {
+    fn to_scale_into(&self, out: &mut Vec<u8>) {
+        // match syntax gets a bit cluttered without u256 literals, falling back to if's
+
+        if self.0 <= 0b0011_1111u64.into() {
+            return out.push(self.0.low_u8() << 2);
+        }
+
+        if self.0 <= 0b0011_1111_1111_1111u64.into() {
+            let v = u16::try_from(self.0).expect("cannot happen as we just checked the value");
+            return ((v << 2) | 0b01).to_scale_into(out);
+        }
+
+        if self.0 <= 0b0011_1111_1111_1111_1111_1111_1111_1111u64.into() {
+            let v = u32::try_from(self.0).expect("cannot happen as we just checked the value");
+            return ((v << 2) | 0b10).to_scale_into(out);
+        }
+
+        let bytes_needed = 32 - self.0.leading_zeros() / 8;
+        out.reserve(bytes_needed as usize);
+        out.push(0b11 + ((bytes_needed - 4) << 2) as u8);
+        let mut x = self.0;
+        for _ in 0..bytes_needed {
+            out.push(x.low_u8());
+            x >>= 8;
+        }
+    }
+}
+
+impl ToScale for Compact<usize> {
+    fn to_scale_into(&self, out: &mut Vec<u8>) {
+        Compact(self.0 as u64).to_scale_into(out)
+    }
+}
+
+impl<T> ToScale for Option<T>
+where
+    T: ToScale,
+{
+    fn to_scale_into(&self, out: &mut Vec<u8>) {
+        if let Some(t) = &self {
+            t.to_scale_into(out);
+        }
+    }
+}
+
+impl<T> ToScale for &[T]
+where
+    T: ToScale,
+{
+    fn to_scale_into(&self, out: &mut Vec<u8>) {
+        Compact(self.len()).to_scale_into(out);
+        for ts in self.iter() {
+            ts.to_scale_into(out);
+        }
+    }
+}
+
+impl<T> ToScale for Vec<T>
+where
+    T: ToScale,
+{
+    fn to_scale_into(&self, out: &mut Vec<u8>) {
+        self.as_slice().to_scale_into(out)
+    }
+}
+
+macro_rules! tuple_impl {
+    ($(,)?) => {};
+    ($first:tt, $($rest:tt),* $(,)?) => {
+        tuple_impl!($($rest),*,);
+        tuple_impl!(@make_impl $first $($rest)*);
+    };
+    (@make_impl $($t:tt)+) => {
+        #[allow(non_snake_case, unused_parens, unconditional_recursion)] // the compiler seems confused here
+        impl <$($t),+> ToScale for ($($t),+,) where $($t: ToScale),+ {
+            fn to_scale_into(&self, out: &mut Vec<u8>) {
+                let ($($t),*) = self;
+                $(
+                    $t.to_scale_into(out);
+                )*
+            }
+        }
+    };
+}
+
+tuple_impl!(T0, T1, T2, T3, T4, T5, T6, T7);
```

Comment:
Do we really need to implement that trait for tuple of 7 items? :D
Tbh I like the macro, but I find it difficult to maintain. Do you think we can just implement the Scale coding for a simple tuple? Do we even need it?